containers/daap-create-schema/src/var/task/create_schema.py

```diff
@@ -118,105 +117,32 @@ def handler(event, context):
             HTTPStatus.BAD_REQUEST, event=event, message=error_msg
         )
 
-    schema = DataProductSchema(
-        data_product_name=data_product_name,
-        table_name=table_name,
-        logger=logger,
-        input_data=input_schema,
-    )
-
-    if schema.exists:
-        error_msg = (
-            f"v1 of this schema for table {table_name} already exists. You can upversion this schema if "
-            "there are changes from v1 using the PUT method of this endpoint. Or if this is a different "
-            "table then please choose a different name for it."
-        )
-        logger.error("create schema called where v1 already exists.")
-        return format_error_response(
-            HTTPStatus.FORBIDDEN, event=event, message=error_msg
-        )
+    version_manager = VersionManager(data_product_name=data_product_name, logger=logger)
 
-    if not schema.has_registered_data_product:
-        error_msg = (
-            f"Schema for {table_name} has no registered metadata for the data product it belongs to. "
-            "Please first register the data product metadata using the POST method of the /data-product/register"
-            " endpoint."
-        )
-        logger.error("schema has no associated registered data product metadata.")
-        return format_error_response(
-            HTTPStatus.FORBIDDEN, event=event, message=error_msg
+    try:
+        new_version, schema = version_manager.create_schema(
+            table_name=table_name, input_data=input_schema
         )
+    except InvalidUpdate as e:
+        error_message = str(e)
+        if error_message.startswith(f"schema for {table_name} has failed validation:"):
```

Comment:
(Non blocking) If we need to distinguish these two cases it would be better to split `InvalidUpdate` into two exceptions, then we can just have something like

```python
except InvalidSchema:
  ...
except InvalidUpdate:
  ...
```

Don't think it needs to block the merge though